custom/ewsghana/alerts/alerts.py

```diff
@@ -0,0 +1,385 @@
+@periodic_task(run_every=crontab(day_of_month=1, hour=10, minute=3),
+               queue=getattr(settings, 'CELERY_PERIODIC_QUEUE', 'celery'))
+def reminder_to_visit_website():
+    domains = EWSGhanaConfig.get_all_enabled_domains()
+    for domain in domains:
+        for user in CommCareUser.by_domain(domain):
+            if user.location and user.last_login < datetime.datetime.now() - datetime.timedelta(weeks=13) and\
+                    user.get_verified_number() and \
+                    (user.location.location_type == 'district' or user.location.location_type == 'region'
+                     or user.location.location_type == 'country'):
+                    message = WEB_REMINDER % user.name
+                    send_sms_to_verified_number(user.get_verified_number(), message)
+                    if user.email:
+                        email = str(user.email)
+                        send_mail('REMINDER TO VISIT WEBSITE', message, 'commcarehq-noreply@dimagi.com', [email])
+
+
+# One week reminder when facility does not report to EWS
+@periodic_task(run_every=crontab(day_of_week=1, hour=11, minute=11),
+               queue=getattr(settings, 'CELERY_PERIODIC_QUEUE', 'celery'))
+def report_reminder():
+    sp_ids = set()
+    now = datetime.datetime.utcnow()
+    date = now - datetime.timedelta(days=7)
+    domains = EWSGhanaConfig.get_all_enabled_domains()
+    for domain in domains:
+        for user in CommCareUser.by_domain(domain):
+            if user.location:
+                sp = SupplyPointCase.get_by_location(user.location)
+                if sp and not StockTransaction.objects.filter(
+                    case_id=sp._id, type="stockonhand", report__date__gte=date).exists()\
+                        and user.get_verified_number():
+                    sp_ids.add(sp._id)
+                    message = REPORT_REMINDER % (user.name, user.location.name)
+                    send_sms_to_verified_number(user.get_verified_number(), message)
+                    if user.email:
+                        email = str(user.email)
+                        send_mail('REPORT REMINDER', message, 'commcarehq-noreply@dimagi.com', [email])
+
+
+# Checking if report was complete or not
+def report_completion_check(user):
+    sp_id = SQLLocation.objects.get(domain=user.domain, location_id=user.location._id).supply_point_id
+    now = datetime.datetime.utcnow()
+    reported_products = StockTransaction.objects.filter(case_id=sp_id, type='stockonhand', report__date=now)
+    expected_products = SQLProduct.objects.filter(domain=user.domain)
+    reported_products_ = set()
+    expected_products_ = set()
+    for product in reported_products:
+        reported_products_.add(SQLProduct.objects.get(product_id=product.product_id).name)
+    for product in expected_products:
+        expected_products_.add(SQLProduct.objects.get(product_id=product.product_id).name)
+    missing_products = set.difference(reported_products_, expected_products_)
```

Comment:
This method is pretty confusing and going to cause an enormous number of unnecessary queries.

`reported_products` is a list of transactions
`expected_products` is a list of `SQLProduct`s that you then iterate over to refetch all `SQLProduct`s in that list and add their names to a set..

You should just get the `expected_products` set by: 
`set(SQLProduct.objects.filter(domain=user.domain).values_list('name', flat=True))` 
which will get the list that you're getting, but only the names, and as a flat list, then convert that whole thing to a set.

Getting the set for `reported_products` is a bit trickier, but you can do basically the same:
`set(StockTransaction.objects.filter(case_id=sp_id, type='stockonhand', report__date=now).values_list('sql_product__name', flat=True))`